numba/targets/linalg.py

```diff
@@ -1691,3 +1691,319 @@ def det_impl(a):
+    # range of values to select different norm types via kwarg `ord`.
+    # The implementation below branches on dimension of the input
+    # (1D or 2D). The default for `ord` is `None` which requires
+    # special handling in numba, this is dealt with first in each of
+    # the dimension branches. Following this the various norms are
+    # computed via code that is in most cases simply a loop version
+    # of a ufunc based version as found in numpy.
+
+    # The following is common to both 1D and 2D cases.
+    # Convert typing floats to numpy floats for use in the impl.
+    # The return type is always a float, numba differs from numpy in
+    # that it returns an input precision specific value whereas numpy
+    # always returns np.float64.
+    nb_ret_type = getattr(a.dtype, "underlying_float", a.dtype)
+    np_ret_type = np_support.as_dtype(nb_ret_type)
+
+    np_dtype = np_support.as_dtype(a.dtype)
+
+    xxnrm2_sig = types.intc(types.char,  # kind
+                            types.intp,  # n
+                            types.CPointer(a.dtype),  # x
+                            types.intp,  # incx
+                            types.CPointer(nb_ret_type))
+
+    xxnrm2 = types.ExternalFunction("numba_xxnrm2", xxnrm2_sig)
+
+    kind = ord(get_blas_kind(a.dtype, "norm"))
+
+    one = 1
+
+    FORCE_CONTIG = a.layout not in 'CF'
+
+    if a.ndim == 1:
+        # 1D cases
+
+        # Used if order is None or order == 2 : see note below
+        @jit(nopython=True)
+        def oneD_none_or_2_impl(a, order=None):
+            # Just ignore order, calls are guarded to only come
+            # from cases where order=None or order=2.
+            n = len(a)
+            # Call L2-norm routine from BLAS
+            ret = np.empty((1,), dtype=np_ret_type)
+            jmp = int(a.strides[0] / a.itemsize)
+            r = xxnrm2(
+                kind,      # kind
+                n,         # n
+                a.ctypes,  # x
+                jmp,       # incx
+                ret.ctypes  # result
+            )
+            if r < 0:
+                fatal_error_func()
+                assert 0   # unreachable
+
+            # help liveness analysis
+            ret.size
+            a.size
+
+            return ret[0]
+
+        # handle "ord" being "None", must be done separately
+        if ord_flag in (None, types.none):
+            oneD_impl = oneD_none_or_2_impl
+        else:
+            @jit(nopython=True)
+            def oneD_impl(a, order=None):
+                n = len(a)
+                # Note: on order == 2
+                # This is the same as for ord=="None" but because
+                # we have to handle "None" specially this condition
+                # is separated
+                if order == 2:
+                    return oneD_none_or_2_impl(a, order=order)
+                elif order == np.inf:
+                    # max(abs(a))
+                    ret = abs(a[-1])
```

Comment:
I see what I was doing. I've altered this to use `a[0]` now and skewed the range from the front, this will probably be quicker anyway as it's not jumping around in memory. I've also made all the various `abs` like calls as `abs`.